src/compiler/checker.ts

```diff
@@ -17909,6 +17914,16 @@ namespace ts {
             diagnostics.add(resultDiagnostic);
         }
 
+        function isStaticPropertyExists(propNode: Identifier, containingType: Type): boolean {
+            if (!(getObjectFlags(containingType) & ObjectFlags.Class)) {

```

Comment:
do you need this check? I think hasModifier(Static) only returns true for properties that are declared on classes, and this is an error case, so the time needed for getTypeOfSymbol and hasModifier is not that much of a concern. In other words, I think you can just

```ts
return prop && prop.valueDeclaration && hasModifier(prop.valueDeclaration, ModifierFlags.Static)
```

Reply:
```typescript
if (!(getObjectFlags(containingType) & ObjectFlags.Class)) {
```

This condition was added in order to check only class properties, and exclude cases like fn.length, fn.name etc. If there are better solutions to this, I would be appreciated if you could give me any ideas. Thanks in advance.

Reply:
I tried the above change myself and ended up with the following code:

```ts
function typeHasStaticProperty(propName: __String, containingType: Type) {
    const prop = containingType.symbol && getPropertyOfType(getTypeOfSymbol(containingType.symbol), propName);
    return prop && prop.valueDeclaration && hasModifier(prop.valueDeclaration, ModifierFlags.Static);
}
```

This actually catches 7 more tests that need to change, so I think it's an improvement.